flake.nix

```
@@ -1,7 +1,46 @@
 {
-  description = "PolyMC";
-
-  outputs = inputs: {
-    overlay = import ./packages/nix/overlay.nix;
+  description = "PolyMC flake";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.libnbtplusplus = {
+    url = "github:multimc/libnbtplusplus";
+    flake = false;
+  };
+  inputs.quazip = {
+    url = "github:multimc/quazip";
+    flake = false;
   };
+
+  outputs = inputs@{ self, nixpkgs, flake-utils, libnbtplusplus, quazip, ... }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" "aarch64-linux" ] (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+        };
+ 
+        packages = {
+          polymc = pkgs.libsForQt5.callPackage ./packages/nix/polymc {
+            inherit self;
+            submoduleQuazip = quazip;
+            submoduleNbt = libnbtplusplus;
+          };
+        };
+
+        overlay = (final: prev: rec { 
+          polymc = packages.polymc;
```

```suggestion
          polymc = final..libsForQt5.callPackage ./packages/nix/polymc { };
```
If I may interject, when a flake contains an overlay, it's generally considered nicer if the overlay pulls its dependencies from the overlaid package set rather than from the flake lock. It avoids duplicating packages and gives the end user more control. And it's still possible to use `packages.$system.polymc` directly if necessary. I cant find anything written down about that but I can point to some popular flakes doing it.

---

Thanks!